lib/Subscriber/Behavior/Path/ExplicitSubscriber.php

```diff
@@ -141,36 +138,74 @@ public function handlePersist(PersistEvent $event)
         $event->setNode($node);
     }
 
-    private function renameNode(NodeInterface $node, $nodeName)
+    /**
+     * Handle the existing node.
+     *
+     * If the path of the parent node in the event is different from the path
+     * of the actual parent node path for the document then we perform an
+     * implicit move.
+     *
+     * If the node name is different from the of the PHPCR node, then we rename
+     * the PHPCR node accordingly.
+     *
+     * Otherwise, there is nothing to do.
+     *
+     * @param NodeManager $nodeManager
+     * @param NodeInterface $node  The existing node.
+     * @param NodeInterface $parentNode  The parent node from the event (not necessarily the subject node)
+     * @param string $nodeName  The target name of the node.
+     */
+    private function handleExisting(NodeManager $nodeManager, NodeInterface $node, NodeInterface $parentNode, $nodeName)
     {
+        $realParentPath = PathHelper::getParentPath($node->getPath());
+        $eventParentPath = $parentNode->getPath();
+
+        if ($eventParentPath !== $realParentPath) {
```

Comment:
It looks like I introduced it early on - guessing it is todo with being able to persist the same document with different paths -- (the commit message was `implicit move`) -- this is required for the synchronization.